src/machine/machinectl.c

```diff
@@ -215,7 +226,12 @@ static int list_machines(int argc, char *argv[], void *userdata) {
                 }
 
                 machines[n_machines].os = NULL;
-                r = get_os_name(bus, name, &machines[n_machines].os);
+                r = get_os_release_property(bus, name, &machines[n_machines].os, "ID");
+                if (r < 0)
+                        goto out;
+
+                machines[n_machines].version_id = NULL;
+                r = get_os_release_property(bus, name, &machines[n_machines].version_id, "VERSION_ID");
```

Comment:
Hmmpf, getting the OS release is far from free. It involves bus communication, forking off a process that joins the namespace, and so on. Please change this so that it queries both fields in one call. After all the bus call will return everything you need, you just need to make use of two of the entries it returns instead of one...